configuration.cpp

```diff
@@ -1,90 +1,106 @@
+			maximum_delay_time = load<int>(cfg, "maximum_delay_time", 30);
 
-                // string parameters
+			// string parameters
 
-                capture_parameter("index", index);
-                capture_parameter("url_template", urlTemplate);
-                capture_parameter("delay_parameters",   delay_parameters);
-                capture_parameter("collection_test_flag",  collection_test_flag);
-            } catch ( const exception& _e ) {
-                THROW( KEY_NOT_FOUND, fmt::format("[{}:{}] - [{}] [error_code=[{}], instance_name=[{}]",
-                                      __func__, __LINE__, _e.client_display_what(), _e.code(), _instance_name));
-            } catch ( const nlohmann::json::exception& _e ) {
-                irods::log( LOG_ERROR,
-                            fmt::format("[{}:{}] in [file={}] - json exception occurred [error={}], [instance_name={}]",
-                                         __func__,__LINE__,__FILE__, _e.what(), _instance_name));
-                THROW( SYS_LIBRARY_ERROR, _e.what() );
-            } catch ( const std::exception& _e ) {
-                THROW( SYS_INTERNAL_ERR,
-                            fmt::format("[{}:{}] in [file={}] - general exception occurred [error={}], [instance_name={}]",
-                                         __func__,__LINE__,__FILE__, _e.what(), _instance_name));
-            } catch ( ... ) {
-                THROW( SYS_UNKNOWN_ERROR,
-                       fmt::format( "[{}:{}] in [file={}], [instance_name={}]",__func__,__LINE__,__FILE__,_instance_name));
-            }
+			capture_parameter("index", index);
+			capture_parameter("url_template", url_template);
+			capture_parameter("delay_parameters", delay_parameters);
+			capture_parameter("collection_test_flag", collection_test_flag);
+		}
+		catch (const exception& _e) {
+			THROW(KEY_NOT_FOUND,
+			      fmt::format("[{}:{}] - [{}] [error_code=[{}], instance_name=[{}]",
+			                  __func__,
+			                  __LINE__,
+			                  _e.client_display_what(),
+			                  _e.code(),
+			                  _instance_name));
+		}
+		catch (const nlohmann::json::exception& _e) {
+			// TODO Why log this and not the others?
+			irods::log(LOG_ERROR,
+			           fmt::format("[{}:{}] in [file={}] - json exception occurred [error={}], [instance_name={}]",
+			                       __func__,
+			                       __LINE__,
+			                       __FILE__,
+			                       _e.what(),
+			                       _instance_name));
+			THROW(SYS_LIBRARY_ERROR, _e.what());
+		}
+		catch (const std::exception& _e) {
+			THROW(SYS_INTERNAL_ERR,
+			      fmt::format("[{}:{}] in [file={}] - general exception occurred [error={}], [instance_name={}]",
+			                  __func__,
+			                  __LINE__,
+			                  __FILE__,
+			                  _e.what(),
+			                  _instance_name));
+		}
```

Comment:
It feels to me like they should be consistent, agreed.

I'm guessing it got a log statement because a lot of debugging has happened for the JSON parsing, and perhaps not so much for the others?

Reply:
I agree.

Reply:
Updated.